src/toil/test/server/wes_client.py

```diff
@@ -0,0 +1,212 @@
+# Modified from wes-service under Apache License, Version 2.0.
+# https://github.com/common-workflow-language/workflow-service/blob/main/LICENSE
```

Comment:
Instead of copy-pasting all this, we could just depend on the `wes-service` package, and e.g. `from wes_client.util import WESClient`, right? What kind of changes do we need to make that would justify reimplementing the bindings here? And can we make them by inheriting most of the code from the other package and just overriding some methods?

Reply:
I talked to @w-gao and we need to duplicate code here because trying to install `wes-service` gets stuck in an infinite loop on a `NameError: name 'Str' is not defined` in the install script for `ruamel.yaml==0.15.77`.

The changes made include support for workflow engine parameters, in a way that's tough to nicely do with inheritance; we'd duplicate a bunch of request building code and end up with a `WESClientWithWorkflowEngineParameters` or something.

So the plan is to keep this in Toil, but contribute the additional features back upstream, and get @mr-c or someone else who has touched `wes-service` to release a version that has these features and also installs. Then we can replace our copy with theirs.